test/servicemesh.go

```diff
@@ -4,12 +4,100 @@ import (
 	"context"
 	"fmt"
 
+	networking "k8s.io/api/networking/v1"
 	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
 	servingv1 "knative.dev/serving/pkg/apis/serving/v1"
 )
 
+const (
+	KnativeSystemNamespaceKey = "knative.openshift.io/system-namespace"
+)
+
+func ServiceMeshControlPlaneV1(name, namespace string) *unstructured.Unstructured {
+	return &unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"apiVersion": "maistra.io/v1",
+			"kind":       "ServiceMeshControlPlane",
+			"metadata": map[string]interface{}{
+				"name":      name,
+				"namespace": namespace,
+			},
+			"spec": map[string]interface{}{
+				"version": "v1.1",
+			},
+		},
+	}
+}
+
+func ServiceMeshControlPlaneV2(name, namespace string) *unstructured.Unstructured {
+	return &unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"apiVersion": "maistra.io/v2",
+			"kind":       "ServiceMeshControlPlane",
+			"metadata": map[string]interface{}{
+				"name":      name,
+				"namespace": namespace,
+			},
+			"spec": map[string]interface{}{
+				"version": "v2.0",
+			},
+		},
+	}
+}
+
+func AddServiceMeshControlPlaneV1IngressGatewaySecretVolume(smcp *unstructured.Unstructured, name, secretName, mountPath string) error {
+	secretVolume := make(map[string]interface{})
+	secretVolume["name"] = name
+	secretVolume["secretName"] = secretName
+	secretVolume["mountPath"] = mountPath
+
+	secretVolumes, found, _ := unstructured.NestedSlice(smcp.Object, "spec", "istio", "gateways", "istio-ingressgateway", "secretVolumes")
```

Comment:
The error from `NestedSlice` is ignored by design? If so, could you add the comment?

Reply:
Fixed a few more of those as well, thanks!